File: src/main/scala/scalademo/syntax/upper_demo.py
class UpperDemo:
    def convert(self, strings):
        for string in strings:
            yield string.upper()


def hello():
    up = UpperDemo()
    strings = ["Hello", "Xander!"]
    result = " ".join(up.convert(strings))
    print(result)


if __name__ == "__main__":
    hello()

File: src/main/scala/scalademo/syntax/UpperDemo.scala
package scalademo.syntax

class UpperDemo:
  def convert(strings: Seq[String]): Seq[String] =
    strings.map((s: String) => s.toUpperCase)

object Demo:
  @main def Hello(): Unit =
    val up = new UpperDemo()
    val strings = List("Hello", "Xander!")
    val result = up.convert(strings).mkString(" ")
    println(result)